src/ds/json/adv-jsonpath.md

```diff
@@ -0,0 +1,191 @@
+
+```redis Invalid path
+JSON.GET obj1 $.c // "[]" is returned
+```
+
+```redis Combining . and []
+JSON.GET arrmap $[0].a
+```
+
+## Bracket (union) expressions
+
+The following syntaxes are supported:
+
+- `'$["value1", "value2", ...]'`
+- `$[idx1, idx2, idx3, ...]`
+
+Each represents a union operation, allowing you to select multiple elements.
+
+For objects, return values are as follows:
+
+- If the current node is an object, an array containing the values of the object’s fields, based on their names, is returned.
+- If the current node is an object and has no members named in the bracket expression, it is skipped.
+This could result in an empty array being returned.
+- If the current node is not an object, an empty array is returned.
+
+Here are some examples:
+
+```redis Bracket union examples with objects
+JSON.GET obj1 '$["a"]'
+JSON.GET obj1 '$["b","b"]'
+JSON.GET obj1 '$["a","c","b","d"]'
+JSON.GET obj1 '$["c","d"]'
+JSON.GET arr1 '$["a","b"]'
+```
+
+For arrays:
+
+- idx must be an integer
+- if idx < 0, it is replaced with min(0, idx + array-length)
+- if idx ≥ array-length (after normalization) - it is skipped
+
+Return values are as follows:
+
+- If the current node is an array, an array containing elements based on one or more 0-based indexes is returned.
+- If the current node is not an array, it is skipped, possibly resulting in an empty array being returned.
+
+idx must be an integer
+if idx < 0, it is replaced with min(0, idx + array-length)
+if idx ≥ array-length (after normalization) - it is skipped
+
+```redis Bracket union examples with arrays
+JSON.GET obj2 $.a[0]
+JSON.GET obj2 $.a[-1]
+JSON.GET obj2 $.a[100]
+JSON.GET obj2 $.a[-100]
+JSON.GET obj2 $.a[8][1]
+JSON.GET obj2 $.a[0,0,1,1]
+JSON.GET obj2 $.b[0]
+JSON.GET obj2 $.*[0]
+```
+
+Redis Stack also supports slice syntax for arrays: `[start:`end`:`step`]`, where `start`, ``end``, and ``step`` are indexes.
+The following rules apply:
+
+- If the current node is an array, an array containing elements extracted from an array are returned, based on a start index, an `end` index, and a `step` index.
+Array indexes are 0-based. Start Index is inclusive; End index is not inclusive.
+The following rules apply:
+
+  - If `start` is specified, it must be an integer.
+  - if `start` is omitted, it is replaced with 0.
+  - if `start` < 0, it is replaced with min(0, `start` + array-length).
+  - if `start` > array-length, an empty array is returned.
+  - if `end` is specified, it must be an integer.
+  - If `end` is omitted, it is replaced with array-length.
+  - If `end` ≥ array-length, it is replaced with array-length.
+  - if `end` < 0, it is replaced with min(0, `end` + array-length).
+  - If `end` ≤ `start` (after normalization), an empty array is returned.
+  - if `step` is specified, it must be a positive integer.
+  - if `step` is omitted, it is replaced with 1.
+  - If the current node in not an array, an empty array is returned.
```

Comment:
just an idea - maybe convert this list into a table? so the rules can be easily compared

Reply:
Not sure that it looks better, but it's done. :)